commit f48f43261ce20779f0d3cd03e6503432fb06f2c9
Author: Tim Kosse <tim.kosse@filezilla-project.org>
Date:   Mon Dec 4 13:15:44 2017 +0100

    It is possible for CGDisplayModeCopyPixelEncoding to return NULL, e.g. during a switch between the built-in screen and an external monitor on a Macbook. Check the return value to prevent a crash.

diff --git a/src/osx/utils_osx.cpp b/src/osx/utils_osx.cpp
index 47e07ba683..ce280f6a5a 100644
--- a/src/osx/utils_osx.cpp
+++ b/src/osx/utils_osx.cpp
@@ -71,16 +71,19 @@ int wxDisplayDepth()
     CGDisplayModeRef currentMode = CGDisplayCopyDisplayMode(kCGDirectMainDisplay);
     CFStringRef encoding = CGDisplayModeCopyPixelEncoding(currentMode);
     
-    if(CFStringCompare(encoding, CFSTR(IO32BitDirectPixels), kCFCompareCaseInsensitive) == kCFCompareEqualTo)
-        theDepth = 32;
-    else if(CFStringCompare(encoding, CFSTR(IO16BitDirectPixels), kCFCompareCaseInsensitive) == kCFCompareEqualTo)
-        theDepth = 16;
-    else if(CFStringCompare(encoding, CFSTR(IO8BitIndexedPixels), kCFCompareCaseInsensitive) == kCFCompareEqualTo)
-        theDepth = 8;
-    else
-        theDepth = 32; // some reasonable default
+    theDepth = 32; // some reasonable default
+    if(encoding)
+    {
+        if(CFStringCompare(encoding, CFSTR(IO32BitDirectPixels), kCFCompareCaseInsensitive) == kCFCompareEqualTo)
+            theDepth = 32;
+        else if(CFStringCompare(encoding, CFSTR(IO16BitDirectPixels), kCFCompareCaseInsensitive) == kCFCompareEqualTo)
+            theDepth = 16;
+        else if(CFStringCompare(encoding, CFSTR(IO8BitIndexedPixels), kCFCompareCaseInsensitive) == kCFCompareEqualTo)
+            theDepth = 8;
+
+        CFRelease(encoding);
+    }
 
-    CFRelease(encoding);
     CGDisplayModeRelease(currentMode);
 
     return theDepth;
